Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -663,35 +663,25 @@ static void registerLogHook()
         if (msg->buffer_sz > 1024)
             return;
 
-        CString logFormat(msg->format);
-        CString logChannel(msg->subsystem);
-        CString logCategory(msg->category);
+        auto logItem = LogData {
+            CString(msg->subsystem),
+            CString(msg->category),
+            type,
+            CString(msg->format),
+            Vector<uint8_t>(msg->buffer, msg->buffer_sz),
+            Vector<uint8_t>(msg->privdata, msg->privdata_sz)
+        };
 
-        Vector<uint8_t> buffer(msg->buffer, msg->buffer_sz);
-        Vector<uint8_t> privdata(msg->privdata, msg->privdata_sz);
+        Locker lock { m_logDataLock };
 
-        static NeverDestroyed<Ref<WorkQueue>> queue(WorkQueue::create("Log Queue", WorkQueue::QOS::Background));
-
-        queue.get()->dispatch([logFormat = WTFMove(logFormat), logChannel = WTFMove(logChannel), logCategory = WTFMove(logCategory), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
-            os_log_message_s msg = { 0 };
-
-            msg.format = logFormat.data();
-            msg.buffer = buffer.data();
-            msg.buffer_sz = buffer.size();
-            msg.privdata = privdata.data();
-            msg.privdata_sz = privdata.size();
-
-            char* messageString = os_log_copy_message_string(&msg);
-            if (!messageString)
-                return;
-            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
-
-            auto connectionID = WebProcess::singleton().networkProcessConnectionID();
-            if (connectionID)
-                IPC::Connection::send(connectionID, Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel.bytesInludingNullTerminator(), logCategory.bytesInludingNullTerminator(), logString, type, getpid()), 0);
+        m_logSet.append(WTFMove(logItem));
+        if (m_logSet.size() < 32)
+            return;
 
-            free(messageString);
-        });
+        auto connectionID = networkProcessConnectionID();
+        if (!connectionID)
+            return;
+        IPC::Connection::send(connectionID, Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(WTFMove(m_logSet), getpid()), 0);
```

Comment:
Should be using std::exchange(), not WTFMove() given that you'll reuse m_logSet later on. WTFMove() is just a cast. This code is fragile because it both:
- Relies on the Vector move constructor to leave the Vector in a reusable & empty state
- Relies on LogOnBehalfOfWebContent to actually adopt the Vector.